packages/gatsby/src/internal-plugins/query-runner/query-runner.js

```diff
@@ -50,11 +50,12 @@ module.exports = async (pageOrLayout, component) => {
   }
 
   // Add the path/layout context onto the results.
-  let contextKey = `pathContext`
   if (!pageOrLayout.path) {
-    contextKey = `layoutContext`
+    result[`layoutContext`] = pageOrLayout.context
+  } else {
+    result[`pathContext`] = pageOrLayout.context // deprecated!
```

Comment:
I think we should just do a breaking change — this will mean loading the same data twice into the browser which could get expensive.

Reply:
👍